cmd/cosign/cli/fulcio/fulcio.go

```diff
@@ -137,13 +137,32 @@ func NewSigner(ctx context.Context, ko options.KeyOpts) (*Signer, error) {
 		}
 	}
 
-	priv, err := cosign.GeneratePrivateKey()
-	if err != nil {
-		return nil, fmt.Errorf("generating cert: %w", err)
-	}
-	signer, err := signature.LoadECDSASignerVerifier(priv, crypto.SHA256)
-	if err != nil {
-		return nil, err
+	var signer signature.SignerVerifier
+	switch {
```

Comment:
This looks quite similar to `sign.SignerFromKeyOpts`. I think I'd rather have `fulcio.NewSigner` take a `SignerVerifier` and handle it there (see my comment in `SignerFromKeyOpts`).